hugegraph-core/src/main/java/com/baidu/hugegraph/job/schema/IndexLabelRemoveCallable.java

```diff
@@ -54,14 +54,20 @@ protected static void removeIndexLabel(HugeGraphParams graph, Id id) {
             // TODO add update lock
             // Set index label to "deleting" status
             schemaTx.updateSchemaStatus(indexLabel, SchemaStatus.DELETING);
-            // Remove index data
-            // TODO: use event to replace direct call
-            graphTx.removeIndex(indexLabel);
-            // Remove label from indexLabels of vertex or edge label
-            removeIndexLabelFromBaseLabel(schemaTx, indexLabel);
-            removeSchema(schemaTx, indexLabel);
-            // Should commit changes to backend store before release delete lock
-            graph.graph().tx().commit();
+            try {
+                // Remove index data
+                // TODO: use event to replace direct call
+                graphTx.removeIndex(indexLabel);
+                // Remove label from indexLabels of vertex or edge label
+                removeIndexLabelFromBaseLabel(schemaTx, indexLabel);
+                removeSchema(schemaTx, indexLabel);
+                // Should commit changes to backend store
+                // before release delete lock
+                graph.graph().tx().commit();
+            } catch (Throwable e) {
+                schemaTx.updateSchemaStatus(indexLabel, SchemaStatus.INVALID);
+                throw e;
+            }
```

Comment:
should update status to INVALID  when failure to remove vertex/edge label?

Reply:
no need. Fail to remove vertex/edge label will lead to partial delete of vertices/edges. vertex/edge label need to be still valid for remaining vertices/edges